### ISSUE TITLE
Terminal not accepting keyboard input in Windows 10

### ISSUE THREAD

**Issue by bsalmeida** [microsoft/terminal#8045]
### Environment
Windows version: 19041.572
Windows Terminal version: 1.3.2651.0

### Steps to reproduce
Launch Terminal
Attempt to type anything

### Expected behavior
Characters show up on the screen.

### Actual behavior
Nothing happens. It's as if I have no keyboard attached.

### Additional info
Using Windows Remote Desktop.
Right-click on mouse pastes text.
No "de-bloating" software installed (neither MSI Afterburner).
"Touch Keyboard and Handwriting Service" is enabled and running.
I've read #4448, and as it suggests, I'm filling out a new issue.

**Comment by zadjii-msft** [comment]
Huh, that's weird, this is the first time I'm hearing of this bug with the Touch Keyboard and Handwriting Service enabled, and also the first time Remote Desktop has been involved.

By any chance can you paste copied text by right-clicking in the Terminal?

I've also got another theory that this is due to the whole scanCode=0 thing, but debugging that is a little more involved, so let's start with the right-clicking

**Comment by bsalmeida** [comment]
@zadjii-msft Yes. I can paste text by right-clicking as I've mentioned in the Additional Info section.

**Comment by BerhorShadowbuckle** [comment]
I have the same issue, I can paste text with a mouse, but I can't type anything with keyboard.

```
Windows Terminal
Version: 1.3.2651.0
```

**Comment by zadjii-msft** [comment]
Ah sorry, I missed that. By any chance, are you using the on-screen keyboard to input text? That's another variable to try and help isolate the problem. 

I'm gonna cc @lhecker, who's been working in this area recently, and also @miniksa who I think actually uses Remote Desktop to his work machine to see if he can repro this as well. 

<hr>

@BerhorShadowbuckle For the record, can you confirm you have the "Touch Keyboard and Handwriting Service" enabled?

**Comment by lhecker** [comment]
@zadjii-msft If it's related to the `scanCode` issues (I bet it is), then it'll probably be fixed with #7900.

**Comment by bsalmeida** [comment]
@zadjii-msft I'm not using the virtual keyboard, but it doesn't work anyway.

**Comment by miniksa** [comment]
> Ah sorry, I missed that. By any chance, are you using the on-screen keyboard to input text? That's another variable to try and help isolate the problem.
> 
> I'm gonna cc @lhecker, who's been working in this area recently, and also @miniksa who I think actually uses Remote Desktop to his work machine to see if he can repro this as well.
> 
> @BerhorShadowbuckle For the record, can you confirm you have the "Touch Keyboard and Handwriting Service" enabled?

I do work all day through Remote Desktop and I do not have an input issue. I do have the "Touch Keyboard and Handwriting Service" enabled at all levels of my remote connection.

**Comment by zadjii-msft** [comment]
I've just tested this out with my remote desktop connection as well and can't repro this on 1.3.2651.0, so there's gotta be something else we're missing here.

**Comment by Quraian** [comment]
I faced the same issue, but I had Touch Keyboard and Handwriting Service  disabled. After enabling it, input only worked after a restart.

**Comment by flyshadowhan** [comment]
same issue 
windows10 upgrade from 20H1 to 20H2 ,after that windows terminal can't input anything.
windows version 19042.572

**Comment by zadjii-msft** [comment]
**@ anyone who comes to this thread because their keyboard input isn't working**: I'm gonna need you to confirm that the "Touch Keyboard and Handwriting Panel" service is running before posting in this thread. You can check that with the command 

```sh
sc query TabletInputService
```

Which should get you something like:
![image](https://user-images.githubusercontent.com/18356694/97710878-55aad480-1a8a-11eb-9e93-bddadf6c4872.png)


If it's not running, and you start it, please make sure to reboot your PC before trying the Terminal again. 

ONLY IF the service is running and keyboard input isn't working should you post in this thread, especially with any details of things that might be unique about your setup. If you don't confirm the service is running, I'm just gonna mark your comment as an off topic dupe of the discussion in #4448.

**Comment by Binary-Vision** [comment]
How do I enable TabletInputService? Also once enabled do i have to reboot or something?

**Comment by lipecss** [comment]
I have the same issue, I can paste text with a mouse, but I can't type anything with keyboard,

**Comment by zadjii-msft** [comment]
If you're seeing that the service is running, could you check if `ctfmon.exe` is also running? Apparently `ctfmon.exe` is the _real_ important process, which the service is just used to resurrect. 

```
> tasklist /v /fi "imagename eq ctfmon.exe" /fo list

Image Name:   ctfmon.exe
PID:          10828
Session Name: Console
Session#:     1
Mem Usage:    26,536 K
Status:       Running
User Name:    REDMOND\migrie
CPU Time:     0:01:05
Window Title: N/A
```

![image](https://user-images.githubusercontent.com/18356694/98955703-d2937080-24c4-11eb-8daf-ae7b2efad705.png)

**Comment by joaolucasete** [comment]
I have the same issue. I checked the proposed services and both are running. 
![image](https://user-images.githubusercontent.com/52172985/99395446-2e903780-28bf-11eb-8795-42e493881c74.png)

The problem is solved with that proposed in discussion #4448 . However, other programs are not functioning properly. Example: microsoft whiteboard

**Comment by lipecss** [comment]
> I have the same issue, I can paste text with a mouse, but I can't type anything with keyboard
> 
> ![image](https://user-images.githubusercontent.com/40566871/98952575-60c62180-24da-11eb-87fa-3970bd2ba18b.png)

The problem Was solved  

Thanks guy

**Comment by ebadger** [comment]
Please confirm that in addition to tabletinputservice running, ctfmon.exe running (with stable PID) that you also have the following registry key set.


C:\Users\ebadger.REDMOND>reg query hklm\software\microsoft\input /v InputServiceEnabled

HKEY_LOCAL_MACHINE\software\microsoft\input
    InputServiceEnabled    REG_DWORD    0x1

**Comment by joaolucasete** [comment]
Everything running, but I can't type on the windows terminal
![image](https://user-images.githubusercontent.com/52172985/101995602-d5db7100-3ca9-11eb-8439-19ce45e642a9.png)

**Comment by joaolucasete** [comment]
if I go to the registers and change the InputServiceEnabled to 0, I can type back into the terminal. However, other windows applications start to have problems.

![image](https://user-images.githubusercontent.com/52172985/101995689-c27cd580-3caa-11eb-8b73-7fd061731727.png)

**Comment by zadjii-msft** [comment]
> if I go to the registers and change the InputServiceEnabled to 0

## NOTE TO FUTURE READERS

Please don't do this. It might fix the Terminal, but it's definitely not the root cause. Changing this in the registry is going to cause far more issues than it will fix. 

## DO NOT DO THIS

**Comment by Jehoel** [comment]
I'm experiencing this issue - `ctfmon.exe` won't stay running - but I noticed this strange behaviour only after enabling `TabletInputService` - it seems the screen area that the OSK would overlay gets the DWM Aero glass/blur - but never renders. Here's a video:

Running Windows 10 2004, `19041.746`.


https://user-images.githubusercontent.com/1693078/106342499-f3fb3180-6255-11eb-8950-f6222b3411fe.MOV

**Comment by TheCoder333** [comment]
This issue is still ongoing ;(

**Comment by lhecker** [comment]
@TheCoder333 Did you make sure that everything is running (ctfmon, etc.) as mentioned previously?

While it's clearly an issue with Windows if ctfmon can't stay running for some, it unfortunately isn't an issue with this project in itself.

Finally the problems described by the submitter of this issue couldn't be reproduced by any member of this project so far.￼
As such (for the purpose of housekeeping) I'd like to take the liberty to close this issue for now.

**Comment by F8ER** [comment]
Looks like the same on the recent [Microsoft.WindowsTerminal_1.11.2921.0](https://github.com/microsoft/terminal/releases/tag/v1.11.2921.0).

![](https://i.imgur.com/auLBAiT.png)

**Comment by zadjii-msft** [comment]
@F8ER 
![image](https://user-images.githubusercontent.com/18356694/145651532-0672c276-7bf2-4e27-bb12-4ec7f57a3f11.png)

**Comment by F8ER** [comment]
> @F8ER ![image](https://user-images.githubusercontent.com/18356694/145651532-0672c276-7bf2-4e27-bb12-4ec7f57a3f11.png)

I'm sorry, but... what's the purpose of the service? Is there a requirement to use a handwriting service instead of a keyboard?

**Comment by zadjii-msft** [comment]
I'm OOF right now but see: https://github.com/microsoft/terminal/wiki/Unable-to-type-in-the-Terminal%3F

**Comment by F8ER** [comment]
> I'm OOF right now but see: https://github.com/microsoft/terminal/wiki/Unable-to-type-in-the-Terminal%3F

Oh dear... This is something extraordinary...

**Comment by zadjii-msft** [comment]
You know, that's right, that doc isn't clear enough. I've gone ahead and added 

```diff
 Are you unable to type in the Windows Terminal, but every other application on your system seems to be running fine? Then you might need to manually enable the "Touch, Keyboard, and Handwriting Panel Service" in the OS. 
+This (admittedly, poorly named) service is in charge of maintaining the modern input stack, which includes _keyboard_ input. 
```

`ctfmon.exe` is responsible for touch and handwriting input, yes, but it's _also_ responsible for _keyboard_ input. It's a poorly named service which is probably the root cause of all these issues. I'm sure if they just named it "Modern Input Service", we wouldn't be having these issues 😝

**Comment by Jehoel** [comment]
@zadjii-msft Ever since a particularly nasty bad experience I had with the original CTFMon that came with Office XP (or was it even Office 2000?) I've made it a point to disable that particular Windows Service and, honestly, I've (honestly, literally, and seriously) _never_ had any problems caused by disabling that service _until I tried Windows Terminal_.

**All other** UWP, WinUI, XAML (and MFC, WinForms, WPF, etc) apps work perfectly fine and handle all input _except_ this Terminal app.

So why does Terminal require it when _no other WinUI/UWP app seems to?

**Comment by zadjii-msft** [comment]
Honestly, we don't _really_ know. All we know is that XAML Islands (the tech stack that's powering the Terminal) _does_. Might be something weird about the way that input gets shuttled between a Win32 message loop and a CoreInputSite message pump. It might be something that magically gets fixed by moving off islands to WinUI 3, it might not (that's looking less possible by the day regardless). The expert in the area is @ebadger, he might have some insights, or be able to loop someone else in who can comment more.

**Comment by BananaMasterz** [comment]
I'm having the same issue (but in windows 11). I noticed that the **Touch Keyboard and Handwriting Panel Service** was disabled. I enabled it and set it to automatic, restarted and all good. BUT THEN the next time I restarted the PC the service didn't start and switched back to manual as seen in the screenshot... now every other restart I have to enable the service and restart....

Edition	Windows 11 Pro
Version	21H2
Installed on	‎11/‎7/‎2022
OS build	22000.795

![image](https://user-images.githubusercontent.com/45745548/178937240-9af47e03-3fa4-4b32-b6bb-247c3f0a1de4.png)

**Comment by rodrigomarquest** [comment]
> I'm having the same issue (but in windows 11). I noticed that the **Touch Keyboard and Handwriting Panel Service** was disabled. I enabled it and set it to automatic, restarted and all good. BUT THEN the next time I restarted the PC the service didn't start and switched back to manual as seen in the screenshot... now every other restart I have to enable the service and restart....
> 
> Edition Windows 11 Pro Version 21H2 Installed on ‎11/‎7/‎2022 OS build 22000.795
> 
> ![image](https://user-images.githubusercontent.com/45745548/178937240-9af47e03-3fa4-4b32-b6bb-247c3f0a1de4.png)

Hello,

Is your workstation standalone or domain joined? do you have any group policy applied (rsop.msc)?
Maybe security tools are disabling some services?

Try putting ctfmon.exe to startup run registry:
REG ADD HKCU\Software\Microsoft\Windows\CurrentVersion\Run /v ctfmon /t REG_SZ /d "C:\Windows\System32\ctfmon.exe" /f

Or copying a ctfmon.exe shortcut to startup folder:
C:\ProgramData\Microsoft\Windows\Start Menu\Programs\StartUp

Checking services in cmd.exe: 
sc query tabletinputservice
sc query hidserv

Configuring services for on demand start:
sc config hidserv tabletinputservice= demand
sc config hidserv start= demand

**Comment by BananaMasterz** [comment]
> > I'm having the same issue (but in windows 11). I noticed that the **Touch Keyboard and Handwriting Panel Service** was disabled. I enabled it and set it to automatic, restarted and all good. BUT THEN the next time I restarted the PC the service didn't start and switched back to manual as seen in the screenshot... now every other restart I have to enable the service and restart....
> > Edition Windows 11 Pro Version 21H2 Installed on ‎11/‎7/‎2022 OS build 22000.795
> 
> Hello,
> 
> Is your workstation standalone or domain joined? do you have any group policy applied (rsop.msc)? Maybe security tools are disabling some services?
> 
> Try putting ctfmon.exe to startup run registry: REG ADD HKCU\Software\Microsoft\Windows\CurrentVersion\Run /v ctfmon /t REG_SZ /d "C:\Windows\System32\ctfmon.exe" /f
> 
> Or copying a ctfmon.exe shortcut to startup folder: C:\ProgramData\Microsoft\Windows\Start Menu\Programs\StartUp
> 
> Checking services in cmd.exe: sc query tabletinputservice sc query hidserv
> 
> Configuring services for on demand start: sc config hidserv tabletinputservice= demand sc config hidserv start= demand

My workstation is a standalone freshly installed windows too. No antivirus or any security settings other than default windows defender settings.

copying a ctfmon.exe shortcut to startup folder used to work 1 year ago but since I clean installed windows 11 it doesn't work anymore. Even when I manually run ctfmon.exe the issue still exists.

Here's a screenshot of the terminal after running the commands:

![image](https://user-images.githubusercontent.com/45745548/187895887-551c896c-50bc-40d5-a721-bb5eced5c440.png)

**Comment by rodrigomarquest** [comment]
> > > I'm having the same issue (but in windows 11). I noticed that the **Touch Keyboard and Handwriting Panel Service** was disabled. I enabled it and set it to automatic, restarted and all good. BUT THEN the next time I restarted the PC the service didn't start and switched back to manual as seen in the screenshot... now every other restart I have to enable the service and restart....
> > > Edition Windows 11 Pro Version 21H2 Installed on ‎11/‎7/‎2022 OS build 22000.795
> > 
> > 
> > Hello,
> > Is your workstation standalone or domain joined? do you have any group policy applied (rsop.msc)? Maybe security tools are disabling some services?
> > Try putting ctfmon.exe to startup run registry: REG ADD HKCU\Software\Microsoft\Windows\CurrentVersion\Run /v ctfmon /t REG_SZ /d "C:\Windows\System32\ctfmon.exe" /f
> > Or copying a ctfmon.exe shortcut to startup folder: C:\ProgramData\Microsoft\Windows\Start Menu\Programs\StartUp
> > Checking services in cmd.exe: sc query tabletinputservice sc query hidserv
> > Configuring services for on demand start: sc config hidserv tabletinputservice= demand sc config hidserv start= demand
> 
> My workstation is a standalone freshly installed windows too. No antivirus or any security settings other than default windows defender settings.
> 
> copying a ctfmon.exe shortcut to startup folder used to work 1 year ago but since I clean installed windows 11 it doesn't work anymore. Even when I manually run ctfmon.exe the issue still exists.
> 
> Here's a screenshot of the terminal after running the commands:
> 

The both services are running but the "sc config..." must be runned in CMD as Administrator.
You can try this commands to check Windows system files:
`sfc /scannow`
`DISM.exe /Online /Cleanup-image /Scanhealth`
`DISM.exe /Online /Cleanup-image /Restorehealth`

Is WSL working?

![image](https://user-images.githubusercontent.com/20219989/187916596-95b16b48-720f-4164-b2fd-f1fcc93182cb.png)

Check Event Viewer either.

**Comment by BananaMasterz** [comment]
> I'm having the same issue (but in windows 11). I noticed that the **Touch Keyboard and Handwriting Panel Service** was disabled. I enabled it and set it to automatic, restarted and all good. BUT THEN the next time I restarted the PC the service didn't start and switched back to manual as seen in the screenshot... now every other restart I have to enable the service and restart....
> 
> Edition Windows 11 Pro Version 21H2 Installed on ‎11/‎7/‎2022 OS build 22000.795
> 
> ![image](https://user-images.githubusercontent.com/45745548/178937240-9af47e03-3fa4-4b32-b6bb-247c3f0a1de4.png)

I did everything in this thread and more and this keeps changing to "manual" instead of "automatic" after every restart... this is infuriating

**Comment by BananaMasterz** [comment]
For anyone interested, I made a workaround since the service **"TabletInputService"** kept going back to manual after every restart.

**Before I start,** try copying **ctfmon.exe** from **_C:\Windows\System32_** to **_C:\Users\...\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Startup_**  and restart your computer. If that doesn't fix your problem, read below.

1) Create a .bat file (name it whatever like **keyboard_workaround.bat**) right click on it to edit and put the following code inside: 

```
sc config TabletInputService start=auto
net start TabletInputService
```

2) save that .bat file somewhere and don't move it ever (I have it in _**C:\Users\...\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Startup**_)

3) Now open **Task Scheduler** and on the right in the **Actions** pane you should see a **Create Task...** option. Select that.

4) In the window that pops, up set whatever name you want (example: keyboard_workaround)

5) Select **Run whether user is logged on or not**

6) Check the checkbox **Run with highest privileges**

7) In **Configure for:** select **windows 10** (or your version if available, as of writing this, no windows 11 option exists so just select 10)

8) On the top select **Triggers**

9) Click **New...** at the bottom

10) At the top where it says **Begin the task:** select **At startup** and click **OK**

11) On the top select **Actions**

12) Click **New...** at the bottom

13) Click **Browse...** in the **Program/script:** field

14) go to the location where you saved your .bat file in step 2 and select it by clicking on it and clicking **open** at the bottom right, then click **OK**

15) Click **OK** and restart your computer.

You should be ready to go.

**IMPORTANT NOTE:** Make sure you check the checkbox **Run with highest privileges** in step 6 !!

**Comment by mikerains** [comment]
Huh, Windows Terminal was working previously, last month, on my Windows 11 Pro Lenovo, now I have this issue, and having done all of the above, still it refuses to accept keyboard input.  Although pressing Ctrl does past my clipboard to the terminal which is weird.

**Comment by mikerains** [comment]
Enabling Touch Keyboard has a huge negative side-effect when running Windows 11 on a laptop that has a touchscreen - the service promiscuously opens the onscreen keyboard whenevr I use the touch screen to scroll a window - even though I have a physical keyboard.  So now I have this on-screen keyboard popping up on top of what I am trying to read and work with.  I have to continually close the OSK.  It's like MS asked "what can we do to interrupt the software developer?  More than just annoy them, we want to really stop them from being productive... hmm, what if we make the OSK pop-up promiscuously? 

So for this reason I am not going to enable Touch Screen service, it makes my laptop UNUSABLE.  I would rather just not use Windows Terminal, I'll just keep using CONEMU.

Who is the manager responsible for making OSK pop-up in laptops that have a Physical Keyboard, can you contact them and get them to understand what a bad requirement this is and get them to at least allow a setting to force OSK to never ever pop-up?

**Comment by mikerains** [comment]
wait ... if this is the new Modern Input Service, and it requires the Touch Screen Keyboard, then when clients start using this for input, nothing will work unless the Touch Screen service is running?  So in order to use Windows, I have to suffer the OSK popping up any time I simply want to scroll a window with my thumb?   If this is really so, it will be the end of Windows for me, I will go exclusively Linux in this case.

**Comment by ebadger** [comment]
> Enabling Touch Keyboard has a huge negative side-effect when running Windows 11 on a laptop that has a touchscreen - the service promiscuously opens the onscreen keyboard whenevr I use the touch screen to scroll a window - even though I have a physical keyboard. So now I have this on-screen keyboard popping up on top of what I am trying to read and work with. I have to continually close the OSK. It's like MS asked "what can we do to interrupt the software developer? More than just annoy them, we want to really stop them from being productive... hmm, what if we make the OSK pop-up promiscuously?
> 
> So for this reason I am not going to enable Touch Screen service, it makes my laptop UNUSABLE. I would rather just not use Windows Terminal, I'll just keep using CONEMU.
> 
> Who is the manager responsible for making OSK pop-up in laptops that have a Physical Keyboard, can you contact them and get them to understand what a bad requirement this is and get them to at least allow a setting to force OSK to never ever pop-up?


If you have a touch screen laptop with a fixed hardware keyboard, OEM should be setting a reg key.  You can set it yourself if you'd like.

https://learn.microsoft.com/en-us/windows-hardware/customize/desktop/unattend/microsoft-windows-gpiobuttons-convertibleslatemode

Do you have this reg key defined?
HKLM/System/CurrentControlSet/Control/PriorityControl/ConvertibleSlateMode

if not, try setting to 1

**Comment by mikerains** [comment]
Hey @ebadger thanks.  That key was set to 0, I changed it to 1, and that does stop the OSK.  This is a relief as I have not been able to find a solution for this since I adopted Win 11 last year, and I was afraid it would be passed over and never solved.  I hope others find this solution, but realy it should be surfaced in Control Panel, on the Touch screen config perhaps.

With this I can leave Touch Screen service running, and not have OSK popup, and Windows Terminal is accepting keystrokes.

Regarding other remedies above, my system's current state is:

My Windows build is Windows 11 21H2 (OS Build 22000.1098)

I added the entry described above, Registry key HKCU\Software\Microsoft\Windows\CurrentVersion\Run
includes an entry with name "ctfmon", type REG_SZ value "C:\Windows\System32\ctfmon.exe"

I do NOT have a shortcut for ctfmon in C:\ProgramData\Microsoft\Windows\Start Menu\Programs\StartUp

$ sc query tabletinputservice

SERVICE_NAME: tabletinputservice
        TYPE               : 30  WIN32
        STATE              : 4  RUNNING
                                (NOT_STOPPABLE, NOT_PAUSABLE, IGNORES_SHUTDOWN)
        WIN32_EXIT_CODE    : 0  (0x0)
        SERVICE_EXIT_CODE  : 0  (0x0)
        CHECKPOINT         : 0x0
        WAIT_HINT          : 0x0

$ sc query hidserv

SERVICE_NAME: hidserv
        TYPE               : 20  WIN32_SHARE_PROCESS
        STATE              : 1  STOPPED
        WIN32_EXIT_CODE    : 1077  (0x435)
        SERVICE_EXIT_CODE  : 0  (0x0)
        CHECKPOINT         : 0x0
        WAIT_HINT          : 0x0

**Comment by jlong-limblecmms** [comment]
I was experiencing issues with the terminal not accepting keyboard or mouse input after resuming from sleep (or whatever the modern equivalent is). In my situation, CTF Loader was taking a lot of CPU. Ending the process in the task manager caused my terminal window to crash, but on launching a new terminal instance it started to accept input again.

**Comment by julie777** [comment]
I also have the problem with Windows Terminal not recognizing keyboard input. It was working fine with the touch keyboard service disabled until I restarted today which installed windows updates.  Now the terminal doesn't see keyboard input.

I find the explanation about the "modern keyboard input" unlikely given that everything else works fine without the touch keyboard service running.

Here are the newly applied updates

![image](https://user-images.githubusercontent.com/5090005/200974732-9106434b-7dc5-4b3e-8980-d92ae6da7887.png)

**Comment by julie777** [comment]
I just started the touche keyboard service and restarted Terminal. It still doesn't accept keyboard input.

**Comment by ebadger** [comment]
> I just started the touche keyboard service and restarted Terminal. It still doesn't accept keyboard input.

do you see ctfmon.exe running in your process list?

**Comment by BananaMasterz** [comment]
> I just started the touche keyboard service and restarted Terminal. It still doesn't accept keyboard input.

Check my comment above. I had the same issue and had to restart every other time I wanted to turn on the PC because starting the service required a restart to be applied.

**Comment by connorferster** [comment]
> I was experiencing issues with the terminal not accepting keyboard or mouse input after resuming from sleep (or whatever the modern equivalent is). In my situation, CTF Loader was taking a lot of CPU. Ending the process in the task manager caused my terminal window to crash, but on launching a new terminal instance it started to accept input again.

I have a similar experience. I will just be working away on the terminal and then, for some reason, something changes and it no longer accepts keyboard input. I am on a desktop machine and using a physical keyboard. 

Restarting Terminal fixes the issue but is a bit frustrating.

**Comment by ASKMAGICSHELL** [comment]
> > Enabling Touch Keyboard has a huge negative side-effect when running Windows 11 on a laptop that has a touchscreen - the service promiscuously opens the onscreen keyboard whenevr I use the touch screen to scroll a window - even though I have a physical keyboard. So now I have this on-screen keyboard popping up on top of what I am trying to read and work with. I have to continually close the OSK. It's like MS asked "what can we do to interrupt the software developer? More than just annoy them, we want to really stop them from being productive... hmm, what if we make the OSK pop-up promiscuously?
> > So for this reason I am not going to enable Touch Screen service, it makes my laptop UNUSABLE. I would rather just not use Windows Terminal, I'll just keep using CONEMU.
> > Who is the manager responsible for making OSK pop-up in laptops that have a Physical Keyboard, can you contact them and get them to understand what a bad requirement this is and get them to at least allow a setting to force OSK to never ever pop-up?
> 
> If you have a touch screen laptop with a fixed hardware keyboard, OEM should be setting a reg key. You can set it yourself if you'd like.
> 
> https://learn.microsoft.com/en-us/windows-hardware/customize/desktop/unattend/microsoft-windows-gpiobuttons-convertibleslatemode
> 
> Do you have this reg key defined? HKLM/System/CurrentControlSet/Control/PriorityControl/ConvertibleSlateMode
> 
> if not, try setting to 1

You save my life
I login just to give you a thumb up

**Comment by eheck-is** [comment]
FWIW, my system has a similar problem in Windows 11 (build 22621.3007) on a convertible notebook after waking up from sleep (not always; might have something to do with how long the system has been sleeping and whether the lid is closed, I'm using an external keyboard and monitor). It's not only affecting Windows Terminal (not accepting keyboard input) but also login (have to press Ctrl-Alt-Del to get the login screen), progress display (doesn't show) and refresh (still shows deleted files) in Explorer, Emoji input panel (opens but doesn't show emoji), for example.
TextInputManagementService is always running and set to automatic startup, This discussion made me try killing ctfmon.exe from Task Manager and that actually helped. ctfmon restarted when I opened a new Windows Terminal and everything seems to be working for now.

**Comment by gerroon** [comment]
The fix doesn't work for me, I use it under RDP (Win 11 to Win 10) The RDP PC (Win 10) has the input service enabled, still I can't type in the command palette.
